File: SpringBoot/src/main/java/com/project/foret/repository/ForetRepository.java
package com.project.foret.repository;

import com.project.foret.entity.Foret;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ForetRepository extends JpaRepository<Foret, Long> {
    List<Foret> findByMembersId(Long id);

    @Query("Select f from Foret f where f.name like %:name%")
    List<Foret> findByContainingName(String name);

    Optional<Foret> findByIdAndMembersId(Long foret_id, Long member_id);
}

File: Android/app/src/main/java/com/project/jjap_guen/repository/ForetRepository.kt
package com.project.jjap_guen.repository

import com.project.jjap_guen.api.RetrofitInstance

class ForetRepository {
    suspend fun getMyForets(member_id: Long) = RetrofitInstance.api.getMyForets(member_id)
    suspend fun getForetBoard(foret_id: Long) = RetrofitInstance.api.getForetBoard(foret_id)
}